### ISSUE TITLE
Sending an existing Qtile keybinding to another program using qtile.cmd_simulate_keypress() fails

### ISSUE THREAD

**Issue by raj-magesh** [qtile/qtile#3040]
I use a terminal (kitty) with tiling-like support and I'd like to reuse the same keybindings I use in Qtile with the kitty panes. However, Qtile has priority over the keybindings. I've managed to disable certain Qtile keybindings whenever a terminal window is focused using this minimal example:

```
def move_focus(qtile, direction):
    if qtile.current_window.get_wm_class()[0] != 'kitty':
        if direction == 'left':
            qtile.current_group.layout.cmd_left()
    else:
        qtile.cmd_simulate_keypress([MOD], 'h')

keys = [Key([MOD], 'h', lazy.function(move_focus, direction='left'), desc='move focus left')]
```

However, the key combination still doesn't get passed through to the terminal. How can I achieve that?

I've found other people with a similar issue:
- https://groups.google.com/g/qtile-dev/c/cp83eFM-JwQ
- https://groups.google.com/g/qtile-dev/c/e25bgej9Ses/m/pAczjD9lmEoJ

_Originally posted by @raj-magesh in https://github.com/qtile/qtile/discussions/3037_

EDIT:

Okay, going through old Qtile issues, the `kitty` developer also raised this issue ~8 years ago: https://github.com/qtile/qtile/issues/324

**Comment by elParaguayo** [comment]
Have a look at #2925 

I think that's fixing the issue that you're looking at as it references the same old issue! 

The PR is in draft but, if you're interested in testing, I'm sure they'd appreciate feedback and/or suggestions.

**Comment by raj-magesh** [comment]
That's awesome, thank you for letting me know! I'm not sure exactly how I'll be able to help out testing, and I'm really busy for the next few weeks, but once I'm free I can download that version and see if my use case works. I'm new to Qtile and collaborative coding, so I don't have much expertise either.

**Comment by elParaguayo** [comment]
No problem at all. 

Help is always appreciated if you're able (there's help in the docs about how to collaborate if you need some tips) but don't worry if you don't have the time.

As you can see, the issue is being looked at so it should get resolved at some point!

**Comment by jwijenbergh** [comment]
This is indeed what #2925 is for, when that is merged we're not completely there yet as actual swallowing based on a function's execution is not yet implemented. However, #2925 supports swallowing based on the functions ```.when(...)``` check. We need to make a separate pr for more `.when()` checks, such as the `wm_class` or even arbitrary `lazy.functions` so if that is merged in a separate pr you can define your keybinding as follows:

```python
Key([MOD], 'h', lazy.layout.left().when(wm_class != 'kitty'), desc='move focus left when kitty is not focused')
```

**Comment by jwijenbergh** [comment]
These custom Match checks have been implemented in #3114

**Comment by jwijenbergh** [comment]
I think this needs to be reopened up until I get the time to finish #2925, which will be soon
